Global AlertmanagerConfig without route spec sends operator into crash loop

**What happened?**

Specifying a global `AlertmanagerConfig` using the new feature introduced in #4220 / d41eb193e5bb250e1a1b0c60a120e4e78ab59765 with an empty route section sends the operator into a crash loop.

**How to reproduce it (as minimally and precisely as possible)**:

1. Create a `AlertManagerConfig` without the `.spec.route` key
2. Create `Alertmanager` and set `.spec.alertManagerConfiguration.name`

* Prometheus Operator version:

`version=0.55.0, branch=refs/tags/v0.55.0, revision=fd0fe0aed99a6a178197f302c06982ad7473a900`

* Kubernetes version information:

```
Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.5", GitCommit:"c285e781331a3785a7f436042c65c5641ce8a9e9", GitTreeState:"clean", BuildDate:"2022-03-16T15:51:05Z", GoVersion:"go1.17.8", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.8", GitCommit:"7061dbbf75f9f82e8ab21f9be7e8ffcaae8e0d44", GitTreeState:"clean", BuildDate:"2022-03-16T14:04:34Z", GoVersion:"go1.16.15", Compiler:"gc", Platform:"linux/amd64"}
```

* Manifests:

```yaml
apiVersion: monitoring.coreos.com/v1
kind: Alertmanager
metadata:
  name: monitoring-kube-prometheus-alertmanager
  namespace: monitoring
spec:
  affinity:
    podAntiAffinity:
      preferredDuringSchedulingIgnoredDuringExecution:
      - podAffinityTerm:
          labelSelector:
            matchExpressions:
            - key: app.kubernetes.io/name
              operator: In
              values:
              - alertmanager
            - key: alertmanager
              operator: In
              values:
              - monitoring-kube-prometheus-alertmanager
          topologyKey: kubernetes.io/hostname
        weight: 100
  alertmanagerConfigNamespaceSelector: {}
  alertmanagerConfigSelector: {}
  alertmanagerConfiguration:
    name: global-alertmanager-config
  image: quay.io/prometheus/alertmanager:v0.24.0
  listenLocal: false
  logFormat: logfmt
  logLevel: info
  paused: false
  portName: http-web
  replicas: 2
  retention: 120h
  routePrefix: /
  securityContext:
    fsGroup: 2000
    runAsGroup: 2000
    runAsNonRoot: true
    runAsUser: 1000
  serviceAccountName: monitoring-kube-prometheus-alertmanager
  version: v0.24.0
---
apiVersion: monitoring.coreos.com/v1alpha1
kind: AlertmanagerConfig
metadata:
  name: global-alertmanager-config
spec:
  inhibitRules:
  - equal: ['prometheus']
    targetMatch:
    - name: namespace
      value: kf-.*
      matchType: =~
    sourceMatch:
    - name: alertname
      value: Watchdog
```

* Prometheus Operator Logs:

```
level=info ts=2022-04-13T15:52:15.233658354Z caller=operator.go:1228 component=prometheusoperator key=monitoring/monitoring-kube-prometheus-prometheus msg="sync prometheus"
level=info ts=2022-04-13T15:52:15.233824382Z caller=operator.go:741 component=alertmanageroperator key=monitoring/monitoring-kube-prometheus-alertmanager msg="sync alertmanager"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x60 pc=0x15f030b]

goroutine 154 [running]:
github.com/prometheus-operator/prometheus-operator/pkg/alertmanager.(*configGenerator).convertRoute(0x1a8e420, 0x0, {{0xc001484fa6, 0x0}, {0xc0014a60c0, 0x0}})
        /home/runner/work/prometheus-operator/prometheus-operator/pkg/alertmanager/amcfg.go:342 +0x8b
github.com/prometheus-operator/prometheus-operator/pkg/alertmanager.(*configGenerator).generateGlobalConfig(0xc002a26568, {0x2138fd0, 0xc0007ce240}, 0xc001736300)
        /home/runner/work/prometheus-operator/prometheus-operator/pkg/alertmanager/amcfg.go:156 +0xcc
github.com/prometheus-operator/prometheus-operator/pkg/alertmanager.(*Operator).provisionAlertmanagerConfiguration(0xc000276c00, {0x2138fd0, 0xc0007ce240}, 0xc0016bfc00, 0xc00042ba40)
        /home/runner/work/prometheus-operator/prometheus-operator/pkg/alertmanager/operator.go:931 +0xa29
github.com/prometheus-operator/prometheus-operator/pkg/alertmanager.(*Operator).sync(0xc000276c00, {0x2138fd0, 0xc0007ce240}, {0xc0001a5d40, 0x32})
        /home/runner/work/prometheus-operator/prometheus-operator/pkg/alertmanager/operator.go:745 +0x655
github.com/prometheus-operator/prometheus-operator/pkg/alertmanager.(*Operator).processNextWorkItem(0xc000276c00, {0x2138fd0, 0xc0007ce240})
        /home/runner/work/prometheus-operator/prometheus-operator/pkg/alertmanager/operator.go:539 +0x132
github.com/prometheus-operator/prometheus-operator/pkg/alertmanager.(*Operator).worker(0xc0004b2840, {0x2138fd0, 0xc0007ce240})
        /home/runner/work/prometheus-operator/prometheus-operator/pkg/alertmanager/operator.go:527 +0x39
created by github.com/prometheus-operator/prometheus-operator/pkg/alertmanager.(*Operator).Run
        /home/runner/work/prometheus-operator/prometheus-operator/pkg/alertmanager/operator.go:462 +0x305
```

**Anything else we need to know?**:

This line https://github.com/prometheus-operator/prometheus-operator/blob/d41eb193e5bb250e1a1b0c60a120e4e78ab59765/pkg/alertmanager/amcfg.go#L156 should probably be

```go
if globalAlertmanagerConfig.Route != nil {
  globalAlertmanagerConfig.Route = cg.convertRoute(amConfig.Spec.Route, crKey)
}
```

--- a/pkg/alertmanager/amcfg.go
+++ b/pkg/alertmanager/amcfg.go
@@ -135,7 +135,9 @@ func (cb *configBuilder) initializeFromAlertmanagerConfig(ctx context.Context, a
 	}
 
 	// Add routes to globalAlertmanagerConfig.Route without enforce namespace
-	globalAlertmanagerConfig.Route = cb.convertRoute(amConfig.Spec.Route, crKey)
+	if amConfig.Spec.Route != nil {
+		globalAlertmanagerConfig.Route = cb.convertRoute(amConfig.Spec.Route, crKey)
+	}
 
 	for _, receiver := range amConfig.Spec.Receivers {
 		receivers, err := cb.convertReceiver(ctx, &receiver, crKey)

--- a/pkg/alertmanager/amcfg.go
+++ b/pkg/alertmanager/amcfg.go
@@ -135,8 +135,8 @@ func (cb *configBuilder) initializeFromAlertmanagerConfig(ctx context.Context, a
 	}
 
 	// Add routes to globalAlertmanagerConfig.Route without enforce namespace
-	if amConfig.Spec.Route != nil {
-		globalAlertmanagerConfig.Route = cb.convertRoute(amConfig.Spec.Route, crKey)
+	if convertedRoute := cb.convertRoute(amConfig.Spec.Route, crKey); convertedRoute != nil {
+		globalAlertmanagerConfig.Route = convertedRoute
 	}
 
 	for _, receiver := range amConfig.Spec.Receivers {
@@ -328,6 +328,9 @@ func (cb *configBuilder) getValidURLFromSecret(ctx context.Context, namespace st
 }
 
 func (cb *configBuilder) convertRoute(in *monitoringv1alpha1.Route, crKey types.NamespacedName) *route {
+	if in == nil {
+		return nil
+	}
 	var matchers []string
 
 	// deprecated

